From 7c7bb3a219f4897e5d681db76e4b32fead5a70da Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Wed, 23 Aug 2023 15:47:20 +0200
Subject: [PATCH 0578/1141] arm64: dts: st: populate all timer counter nodes on
 stm32mp25

Counter driver originally had support limited to quadrature interface and
simple counter. It has evolved, so add remaining counter nodes in stm32
timer nodes.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I47a16b27091e8c47ea33208512c520f49c3cbde9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/330632
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 50 ++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -515,6 +515,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				timer@5 {
 					compatible = "st,stm32mp25-timer-trigger";
 					reg = <5>;
@@ -535,6 +540,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				timer@6 {
 					compatible = "st,stm32mp25-timer-trigger";
 					reg = <6>;
@@ -555,6 +565,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -581,6 +596,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -607,6 +627,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -993,6 +1018,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -1019,6 +1049,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -1174,6 +1209,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -1200,6 +1240,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
@@ -1226,6 +1271,11 @@
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32mp25-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32mp25-pwm";
 					#pwm-cells = <3>;
